[038] create endpoints form internship

Maksim-maevsky opened the pull request:
- added searching method for user by internship id and role
-  added relations many to many to user internship
-  added initialization user_internship table
- added table user_internship

Maksim-maevsky committed:
added endpoint for getting forms by internship id

Maksim-maevsky committed:
added logs for form services and JPQL

Maksim-maevsky committed:
added changelog and init for user-internship relation

Maksim-maevsky committed:
added jpql

Maksim-maevsky committed:
added second parameter type of user to controller

Fenydude committed:
access to all user

Maksim-maevsky committed:
added logs to user service.
added requestParam to user controller

Maksim-maevsky committed:
Merge branch 'development' into 038-create-endpoints-form-internship

# Conflicts:
#	src/main/java/by/exadel/internship/repository/UserRepository.java
#	src/main/java/by/exadel/internship/service/UserService.java
#	src/main/resources/db/changelog/v.0.4/db.changelog-master.yaml

Fenydude committed:
swagger authorize

SergeyKoval commented on src/main/java/by/exadel/internship/controller/FormController.java:
```diff
@@ -31,10 +32,9 @@ public FormFullDTO addNewForm(@RequestPart(name = "form") FormRegisterDTO form,
-    @ResponseStatus(HttpStatus.OK)
-    public List<FormFullDTO> getAllForms() {
-        return formService.getAll();
+    @GetMapping("/{internshipId}")
```
internshipId should be query parameter, not part of the path

Maksim-maevsky replied:
done

SergeyKoval commented on src/main/java/by/exadel/internship/controller/UserController.java:
```diff
@@ -27,9 +25,16 @@
     }
 
-    @GetMapping("/{userId}")
+    @GetMapping("/userId")
```
userId should be query parameter, not part of the path

Maksim-maevsky replied:
sory, maybe I didn't fully understand what you meant about all the cases, now I have fixed everything, done

SergeyKoval replied:
Why do you leave /userId as part of the path? 
Please review https://medium.com/@mwaysolutions/10-best-practices-for-better-restful-api-cbe81b06f291 once again

SergeyKoval commented on src/main/java/by/exadel/internship/controller/UserController.java:
```diff
@@ -27,9 +25,16 @@
         return userService.getById(userId);
     }
+
+    @GetMapping("/internshipId")
```
internshipId should be query parameter, not part of the path

Maksim-maevsky replied:
done

SergeyKoval commented on src/main/java/by/exadel/internship/entity/Form.java:
```diff
@@ -82,4 +82,7 @@
     private List<TimeForCall> timeForCallList;
 
+    @Column(name = "fm_inship_id")
+    private UUID internship;
```
may be better rename it to internshipId as long as it is not an internship entity

Maksim-maevsky replied:
done

Maksim-maevsky committed:
Merge remote-tracking branch 'origin/access-to-resources' into 038-create-endpoints-form-internship

mortal-gpt commented on src/main/java/by/exadel/internship/entity/Internship.java:
```diff
@@ -65,9 +66,14 @@
+            name = "user_internship",
+            joinColumns = @JoinColumn(name = "ui_inship_id"),
+            inverseJoinColumns = @JoinColumn(name = "ui_u_id"))
+    @Type(type = "by.exadel.internship.mapper.enum_mapper.EnumTypePostgreSQL")
```
Does @Type make sense here?

Maksim-maevsky replied:
now, i gues - no, but when we will conection users to internship use FE it will bee usefull.

Maksim-maevsky replied:
yaa i got it, thanks

mortal-gpt commented on src/main/java/by/exadel/internship/entity/User.java:
```diff
@@ -38,15 +41,18 @@
-//    private List<UserInternship> listOfInternships;
-//    private List<LocalDateTime> freeInterviewDates;
+    @ManyToMany(mappedBy = "users", fetch = FetchType.LAZY)
+    @Type(type = "by.exadel.internship.mapper.enum_mapper.EnumTypePostgreSQL")
```
Does @Type annotation make sense here?

Maksim-maevsky replied:
no, i removed it

Maksim-maevsky committed:
access upload

Maksim-maevsky committed:
Merge branch 'development' into 038-create-endpoints-form-internship

# Conflicts:
#	src/main/resources/application.yaml

Maksim-maevsky committed:
added query parameter for each endpoint

Maksim-maevsky committed:
rename var internship to Form

Maksim-maevsky committed:
removed type for entity)

Maksim-maevsky committed:
removed getAll for user controller because we will not get all users without role or skills

when we want to get some object by id, in same Controller , we used path variable.

Example If we want to get sort list something, we use query params

Maksim-maevsky committed:
Merge branch 'development' into 038-create-endpoints-form-internship

# Conflicts:
#	src/main/java/by/exadel/internship/controller/InternshipController.java
#	src/main/java/by/exadel/internship/controller/UserController.java
#	src/main/java/by/exadel/internship/entity/User.java
#	src/main/java/by/exadel/internship/repository/FormRepository.java
#	src/main/java/by/exadel/internship/repository/InternshipRepository.java
#	src/main/java/by/exadel/internship/repository/UserRepository.java
#	src/main/java/by/exadel/internship/service/UserService.java
#	src/main/java/by/exadel/internship/service/impl/FormServiceImpl.java
#	src/main/java/by/exadel/internship/service/impl/UserServiceImpl.java

Maksim-maevsky committed:
structured code

Maksim-maevsky committed:
recovered lost files about fk-internship-form

SergeyKoval commented on src/main/java/by/exadel/internship/controller/UserController.java:
```diff
@@ -19,22 +19,33 @@
-    }
-
-    @GetMapping("/userId")
+    @GetMapping("{userId}")
```
Add / as url prefix

Maksim-maevsky replied:
done

SergeyKoval commented on src/main/java/by/exadel/internship/exception_handing/IncorrectData.java:
```diff
@@ -16,6 +16,7 @@
     public IncorrectData() {
-        this.errorCode = UUID.randomUUID();;
+        this.errorCode = UUID.randomUUID();
+        ;
```
What is that?

Maksim-maevsky replied:
it s for generation exception code error for searching this one in logfile

SergeyKoval replied:
I'm asking about line 20 =)

Maksim-maevsky replied:
yaaa,  thanks)) i removed it.

Maksim-maevsky committed:
correct controller for user
